specification/batch/Azure.Batch/routes.tsp

```diff
@@ -0,0 +1,2237 @@
+the Batch service starts resizing the Pool to zero Compute Nodes; any Tasks
+running on existing Compute Nodes are terminated and requeued (as if a resize
+Pool operation had been requested with the default requeue option); finally,
+the Pool is removed from the system. Because running Tasks are requeued, the
+user can rerun these Tasks by updating their Job to target a different Pool.
+The Tasks can then run on the new Pool. If you want to override the requeue
+behavior, then you should call resize Pool explicitly to shrink the Pool to
+zero size before deleting the Pool. If you call an Update, Patch or Delete API
+on a Pool in the deleting state, it will fail with HTTP status code 409 with
+error code PoolBeingDeleted.
+""")
+  @route("/pools/{poolId}")
+  @projectedName("java", "deletePoolInternal")
+  deletePool is Ops.DeleteOperation<BatchPoolHeaders, DeleteResponseHeaders>;
+
+  @doc("Gets basic properties of a Pool.")
+  @route("/pools/{poolId}")
+  @projectedName("java", "poolExistsInternal")
+  poolExists is Ops.ResourceInfoOperationWithAdditionalResponse<
+    BatchClientRequestHeaders & BatchPoolHeaders,
+    PoolDoesNotExistResponseHeaders,
+    PoolDoesExistResponseHeaders
+  >;
+
+  @doc("Gets information about the specified Pool.")
+  @route("/pools/{poolId}")
+  @projectedName("java", "getPoolInternal")
+  getPool is Ops.ReadOperation<
+    BatchPoolHeaders & {
+      @doc("An OData $select clause.")
+      @query({
+        format: "csv",
+      })
+      @projectedName("client", "select")
+      $select?: string[];
+
+      @doc("An OData $expand clause.")
+      @query({
+        format: "csv",
+      })
+      @projectedName("client", "expand")
+      $expand?: string[];
+    },
+    BatchPool
+  >;
+
+  @summary("Updates the properties of the specified Pool.")
+  @doc("""
+This only replaces the Pool properties specified in the request. For example,
+if the Pool has a StartTask associated with it, and a request does not specify
+a StartTask element, then the Pool keeps the existing StartTask.
+""")
+  @route("/pools/{poolId}")
+  @projectedName("java", "updatePoolInternal")
+  updatePool is Ops.UpdateOperation<
+    BatchPoolHeaders & {
+      @doc("The pool properties to update.")
+      @body
+      pool: BatchPoolUpdateParameters;
+    },
+    RequestSuccesResponseHeaders
+  >;
+
+  @summary("Disables automatic scaling for a Pool.")
+  @doc("Disables automatic scaling for a Pool.")
+  @route("/pools/{poolId}/disableautoscale")
+  @projectedName("java", "disablePoolAutoScaleInternal")
+  disablePoolAutoScale is Ops.PostOperation<
+    {
+      @doc("The ID of the Pool on which to disable automatic scaling.")
+      @path
+      poolId: string;
+    },
+    {
+      @doc("A process exit code.")
+      @statusCode
+      code: "200";
+
+      @doc("The OData ID of the resource to which the request applied.")
+      @header("DataServiceId")
+      dataServiceId: string;
```

Comment:
nit, this head is used multiple times, may worth an alias.

Reply:
made alias and updated